test/test_cpp_bin_float.cpp

```diff
@@ -89,24 +93,24 @@ void test_special_cases()
    BOOST_CHECK_EQUAL(max_val * -2, -inf_val);
    BOOST_CHECK_EQUAL(max_val / half, inf_val);
    BOOST_CHECK_EQUAL(max_val / -half, -inf_val);
-   BOOST_CHECK_EQUAL(max_val / min_val, inf_val);
-   BOOST_CHECK_EQUAL(max_val / -min_val, -inf_val);
+   BOOST_CHECK_EQUAL(max_val / denorm_min_val, inf_val);
+   BOOST_CHECK_EQUAL(max_val / -denorm_min_val, -inf_val);
    // Underflow:
-   BOOST_CHECK_EQUAL(min_val * 2 - one_point_5 * min_val, 0);
-   BOOST_CHECK_EQUAL(-min_val * 2 + one_point_5 * min_val, 0);
-   BOOST_CHECK_EQUAL(min_val / 2, 0);
-   BOOST_CHECK_EQUAL(min_val / max_val, 0);
-   BOOST_CHECK_EQUAL(min_val * half, 0);
-   BOOST_CHECK_EQUAL(min_val - min_val, 0);
+   BOOST_CHECK_EQUAL(denorm_min_val * 2 - one_point_5 * denorm_min_val, 0);
+   BOOST_CHECK_EQUAL(-denorm_min_val * 2 + one_point_5 * denorm_min_val, 0);
+   BOOST_CHECK_EQUAL(denorm_min_val / 2, 0);
+   BOOST_CHECK_EQUAL(denorm_min_val / max_val, 0);
+   BOOST_CHECK_EQUAL(denorm_min_val * half, 0);
+   BOOST_CHECK_EQUAL(denorm_min_val - denorm_min_val, 0);
    BOOST_CHECK_EQUAL(max_val - max_val, 0);
-   BOOST_CHECK_EQUAL(-min_val + min_val, 0);
+   BOOST_CHECK_EQUAL(-denorm_min_val + denorm_min_val, 0);
    BOOST_CHECK_EQUAL(-max_val + max_val, 0);
    // Things which should not over/underflow:
-   BOOST_CHECK_EQUAL((min_val * 2) / 2, min_val);
+   BOOST_CHECK_EQUAL((denorm_min_val * 2) / 2, denorm_min_val);
    BOOST_CHECK_EQUAL((max_val / 2) * 2, max_val);
-   BOOST_CHECK_GE((min_val * 2.0000001) / 1.9999999999999999, min_val);
+   BOOST_CHECK_GE((denorm_min_val * 2.0000001) / 1.9999999999999999, denorm_min_val);
    BOOST_CHECK_LE((max_val / 2.0000001) * 1.9999999999999999, max_val);
-   BOOST_CHECK_EQUAL(min_val * 2 - min_val, min_val);
+   BOOST_CHECK_EQUAL(denorm_min_val * 2 - denorm_min_val, denorm_min_val);
```

Comment:
From lines 99-109 again I would test both min and denorm_min?

Reply:
OK